usr/src/uts/armv8/os/ddi_impl.c

```diff
@@ -989,40 +1303,64 @@ impl_ddi_sunbus_removechild(dev_info_t *dip)
 	impl_rem_dev_props(dip);
 }
 
+/*
+ * Copy name to property_name, since name
+ * is in the low address range below kernelbase.
+ */
+static void
+copy_boot_str(const char *boot_str, char *kern_str, int len)
+{
+	int i = 0;
+
+	while (i < len - 1 && boot_str[i] != '\0') {
+		kern_str[i] = boot_str[i];
+		i++;
+	}
+
+	kern_str[i] = 0;	/* null terminate */
+	if (boot_str[i] != '\0')
+		cmn_err(CE_WARN,
+		    "boot property string is truncated to %s", kern_str);
+}
+
 static void
 get_boot_properties(void)
 {
 	extern char hw_provider[];
 	dev_info_t *devi;
 	char *name;
-	int length;
-	char property_name[OBP_MAXPROPNAME], property_val[50];
+	int length, flags;
+	char property_name[50], property_val[50];
 	void *bop_staging_area;
 
 	bop_staging_area = kmem_zalloc(MMU_PAGESIZE, KM_NOSLEEP);
 
-	devi = ddi_root_node();
-	property_name[0] = '\0';
-	for (name = BOP_NEXTPROP(bootops, property_name);	/* get first */
-	    name && strlen(name) > 0;				/* NULL => DONE */
-	    name = BOP_NEXTPROP(bootops, property_name)) {	/* get next */
+	/*
+	 * Import "root" properties from the boot.
+	 *
+	 * We do this by invoking BOP_NEXTPROP until the list
+	 * is completely copied in.
+	 */
 
-		if (strlen(name) >= OBP_MAXPROPNAME) {
-			cmn_err(CE_NOTE,
-			    "boot property name %s longer than 0x%lx\n",
-			    name, sizeof(property_name));
-			break;
-		}
+	devi = ddi_root_node();
+	for (name = BOP_NEXTPROP(bootops, "");		/* get first */
+	    name;					/* NULL => DONE */
+	    name = BOP_NEXTPROP(bootops, name)) {	/* get next */
 
-		strcpy(property_name, name);
+		/* copy string to memory above kernelbase */
+		copy_boot_str(name, property_name, 50);
 
+		/*
+		 * Skip vga properties. They will be picked up later
+		 * by get_vga_properties.
+		 */
 		if (strcmp(property_name, "display-edif-block") == 0 ||
 		    strcmp(property_name, "display-edif-id") == 0) {
 			continue;
 		}
 
 		length = BOP_GETPROPLEN(bootops, property_name);
-		if (length == 0)
+		if (length < 0)
```

Comment:
What does `length == 0` end up doing?

Reply:
Boolean properties, I meant this as a note to myself to double check and then forgot to delete it.  Sorry.